src/database.py

```diff
@@ -41,6 +42,29 @@ def _query(self, connection, media_id):
                 )
             return (row[0], row[1])
 
+    def set_mediafile(self, media_id, media, mimetype):
+        while True:
+            try:
+                connection = self.get_connection()
+                with connection:
+                    return self._insert(connection, media_id, media, mimetype)
```

Comment:
You are not returning anything from `_insert`. Please use a break-statement outside of the with-clause.

Reply:
Now using a break statement, to exit the set_mediafile function.